Utils/get_private_build_status.py

```diff
@@ -0,0 +1,88 @@
+import os
+import sys
+import json
+import time
+import argparse
+import requests
+import logging
+from urllib.parse import urljoin
+
+# disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+WORKFLOW_HTML_URL = 'https://github.com/demisto/content-private/actions/runs'
+GET_WORKFLOW_URL = 'https://api.github.com/repos/demisto/content-private/actions/runs/'
+
+PRIVATE_REPO_WORKFLOW_ID_FILE = 'PRIVATE_REPO_WORKFLOW_ID.txt'
+
+TIMEOUT_THRESHOLD = 3600  # one hour
+
+
+def get_workflow_status(bearer_token, workflow_id):
+    workflow_url = urljoin(GET_WORKFLOW_URL, workflow_id)
+    res = requests.request("GET",
+                           workflow_url,
+                           headers={'Authorization': bearer_token},
+                           verify=False)
+    if res.status_code != 200:
+        logging.error(
+            f'Failed to gets private repo workflow, request to {workflow_url} failed with error: {str(res.content)}')
+        sys.exit(1)
```

Comment:
In the body of this request you have the key `jobs_url`, you can query this url and get the job's statuses, example output will be:
```
{
  "total_count": 1,
  "jobs": [
    {
      "id": 2010759197,
      "run_id": 613093820,
      "run_url": "https://api.github.com/repos/demisto/content-private/actions/runs/613093820",
      "node_id": "MDg6Q2hlY2tSdW4yMDEwNzU5MTk3",
      "head_sha": "cf08f08c61be1dfebfa9b7cefe8b02d772f32158",
      "url": "https://api.github.com/repos/demisto/content-private/actions/jobs/2010759197",
      "html_url": "https://github.com/demisto/content-private/runs/2010759197",
      "status": "completed",
      "conclusion": "cancelled",
      "started_at": "2021-03-02T06:29:03Z",
      "completed_at": "2021-03-02T06:33:23Z",
      "name": "build",
      "steps": [
        {
          "name": "Set up job",
          "status": "completed",
          "conclusion": "success",
          "number": 1,
          "started_at": "2021-03-02T06:29:03.000+00:00",
          "completed_at": "2021-03-02T06:29:06.000+00:00"
        },
        {
          "name": "first checkout",
          "status": "completed",
          "conclusion": "success",
          "number": 2,
          "started_at": "2021-03-02T06:29:06.000+00:00",
          "completed_at": "2021-03-02T06:29:07.000+00:00"
        },
        {
          "name": "Setup Python",
          "status": "completed",
          "conclusion": "success",
          "number": 3,
          "started_at": "2021-03-02T06:29:07.000+00:00",
          "completed_at": "2021-03-02T06:29:24.000+00:00"
        },
        {
          "name": "install build dependencies",
          "status": "completed",
          "conclusion": "cancelled",
          "number": 4,
          "started_at": "2021-03-02T06:29:24.000+00:00",
          "completed_at": "2021-03-02T06:33:00.000+00:00"
        },
        {
          "name": "install pip and requirements",
          "status": "completed",
          "conclusion": "success",
          "number": 5,
          "started_at": "2021-03-02T06:33:00.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Save current branch in env",
          "status": "completed",
          "conclusion": "skipped",
          "number": 6,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "get content branch",
          "status": "completed",
          "conclusion": "skipped",
          "number": 7,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "content checkout",
          "status": "completed",
          "conclusion": "skipped",
          "number": 8,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Prepare Environment",
          "status": "completed",
          "conclusion": "failure",
          "number": 9,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Save full repo path and token in env",
          "status": "completed",
          "conclusion": "success",
          "number": 10,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Validate pack name",
          "status": "completed",
          "conclusion": "failure",
          "number": 11,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Validate private metadata file",
          "status": "completed",
          "conclusion": "success",
          "number": 12,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "repo checkout",
          "status": "completed",
          "conclusion": "failure",
          "number": 13,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Checkout commit hash or branch",
          "status": "completed",
          "conclusion": "failure",
          "number": 14,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Prepare Venv",
          "status": "completed",
          "conclusion": "failure",
          "number": 15,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Validate Files and Yaml",
          "status": "completed",
          "conclusion": "failure",
          "number": 16,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Validate price exists in pack metadata",
          "status": "completed",
          "conclusion": "failure",
          "number": 17,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "add private label",
          "status": "completed",
          "conclusion": "failure",
          "number": 18,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "add certified label",
          "status": "completed",
          "conclusion": "failure",
          "number": 19,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "add vendor details to pack metadata",
          "status": "completed",
          "conclusion": "failure",
          "number": 20,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Update Tests step",
          "status": "completed",
          "conclusion": "failure",
          "number": 21,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Create ID Set",
          "status": "completed",
          "conclusion": "failure",
          "number": 22,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Infrastructure testing",
          "status": "completed",
          "conclusion": "failure",
          "number": 23,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Configure Test Filter",
          "status": "completed",
          "conclusion": "failure",
          "number": 24,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Create Content Artifacts",
          "status": "completed",
          "conclusion": "failure",
          "number": 25,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Calculate Packs Dependencies",
          "status": "completed",
          "conclusion": "failure",
          "number": 26,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:16.000+00:00"
        },
        {
          "name": "Run Unit Testing and Lint",
          "status": "completed",
          "conclusion": "failure",
          "number": 27,
          "started_at": "2021-03-02T06:33:16.000+00:00",
          "completed_at": "2021-03-02T06:33:17.000+00:00"
        },
        {
          "name": "content test conf checkout",
          "status": "completed",
          "conclusion": "success",
          "number": 28,
          "started_at": "2021-03-02T06:33:17.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Download Configuration",
          "status": "completed",
          "conclusion": "failure",
          "number": 29,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Prepare content packs for testing",
          "status": "completed",
          "conclusion": "failure",
          "number": 30,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Zip Marketplace Packs",
          "status": "completed",
          "conclusion": "failure",
          "number": 31,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Create Instance",
          "status": "completed",
          "conclusion": "failure",
          "number": 32,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Install Oregon SSH key",
          "status": "completed",
          "conclusion": "failure",
          "number": 33,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Setup SSH Keys and known_hosts",
          "status": "completed",
          "conclusion": "failure",
          "number": 34,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Open ssh tunnel",
          "status": "completed",
          "conclusion": "failure",
          "number": 35,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Setup Instance",
          "status": "completed",
          "conclusion": "failure",
          "number": 36,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Run Tests - Latest GA",
          "status": "completed",
          "conclusion": "failure",
          "number": 37,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:22.000+00:00"
        },
        {
          "name": "Create snapshot of git branch",
          "status": "completed",
          "conclusion": "failure",
          "number": 38,
          "started_at": "2021-03-02T06:33:22.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Upload Packs To Marketplace Storage",
          "status": "completed",
          "conclusion": "failure",
          "number": 39,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Pack Upload Information and URLs",
          "status": "completed",
          "conclusion": "failure",
          "number": 40,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Upload artifacts",
          "status": "completed",
          "conclusion": "success",
          "number": 41,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Post content test conf checkout",
          "status": "completed",
          "conclusion": "success",
          "number": 80,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Post repo checkout",
          "status": "completed",
          "conclusion": "success",
          "number": 81,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Post first checkout",
          "status": "completed",
          "conclusion": "success",
          "number": 82,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        },
        {
          "name": "Complete job",
          "status": "completed",
          "conclusion": "success",
          "number": 83,
          "started_at": "2021-03-02T06:33:23.000+00:00",
          "completed_at": "2021-03-02T06:33:23.000+00:00"
        }
      ],
      "check_run_url": "https://api.github.com/repos/demisto/content-private/check-runs/2010759197"
    }
  ]
}
```

Lets use that to better display the status of the workflow i.e jobs statuses, (which jobs ran what was their result, how long they took etc..)